packages/legend-application-query/src/components/Core_LegendQueryApplicationPlugin.tsx

```diff
@@ -727,4 +761,223 @@ export class Core_LegendQueryApplicationPlugin extends LegendQueryApplicationPlu
       },
     };
   }
+
+  override getExtraQueryGraphBuilderGetters(): QueryGraphBuilderGetter[] {
+    return [
+      (
+        editorStore: QueryEditorStore,
+      ): ((editorStore: QueryEditorStore) => GeneratorFn<void>) | undefined => {
+        function* buildGraph(): GeneratorFn<void> {
+          // do nothing
+        }
+        if (editorStore instanceof ExistingQueryEditorStore) {
+          const query = editorStore.query;
+          const dataSpaceTaggedValue = query?.taggedValues?.find(
+            (taggedValue) =>
+              taggedValue.profile === QUERY_PROFILE_PATH &&
+              taggedValue.tag === QUERY_PROFILE_TAG_DATA_SPACE &&
+              isValidFullPath(taggedValue.value),
+          );
+          if (dataSpaceTaggedValue) {
+            return buildGraph;
+          }
+        }
+        return undefined;
+      },
+    ];
+  }
+
+  getExtraQueryBuilderPropagateExecutionContextChangeHelper?(): QueryBuilderPropagateExecutionContextChangeHelper[] {
```

Comment:
HEADS-UP: create an extension method for `propagateExecutionContextChange` because for dataspace, it depends on editorStore (@finos/legend-application-query) and dataspaceQueryBuilderState (@finos/legend-extension-dsl-data-space). 
https://github.com/finos/legend-studio/pull/2523 put this function's implementaion at  module @finos/legend-extension-dsl-data-space, but it will introduce the circular dependency.